packages/rpc-methods/src/restricted/manageState.test.ts

```diff
@@ -0,0 +1,208 @@
+    it('gets snap state', async () => {
+      const clearSnapState = jest.fn().mockResolvedValueOnce(true);
+      const getSnapState = jest.fn().mockResolvedValueOnce(true);
+      const updateSnapState = jest.fn().mockResolvedValueOnce(true);
+
+      const manageStateImplementation = getManageStateImplementation({
+        clearSnapState,
+        getSnapState,
+        updateSnapState,
+      });
+
+      await manageStateImplementation({
+        context: { origin: 'snap-origin' },
+        method: '',
+        params: { operation: ManageStateOperation.getState },
+      });
+
+      expect(getSnapState).toHaveBeenCalledWith('snap-origin');
+    });
+
+    it('clears snap state', async () => {
+      const clearSnapState = jest.fn().mockResolvedValueOnce(true);
+      const getSnapState = jest.fn().mockResolvedValueOnce(true);
+      const updateSnapState = jest.fn().mockResolvedValueOnce(true);
+
+      const manageStateImplementation = getManageStateImplementation({
+        clearSnapState,
+        getSnapState,
+        updateSnapState,
+      });
+
+      await manageStateImplementation({
+        context: { origin: 'snap-origin' },
+        method: '',
+        params: { operation: ManageStateOperation.clearState },
+      });
+
+      expect(clearSnapState).toHaveBeenCalledWith('snap-origin');
+    });
+
+    it('updates snap state', async () => {
+      const clearSnapState = jest.fn().mockResolvedValueOnce(true);
+      const getSnapState = jest.fn().mockResolvedValueOnce(true);
+      const updateSnapState = jest.fn().mockResolvedValueOnce(true);
+
+      const manageStateImplementation = getManageStateImplementation({
+        clearSnapState,
+        getSnapState,
+        updateSnapState,
+      });
+      const newState = { data: 'updated data' };
+
+      await manageStateImplementation({
+        context: { origin: 'snap-origin' },
+        method: '',
+        params: {
+          operation: ManageStateOperation.updateState,
+          newState,
+        },
+      });
+
+      expect(updateSnapState).toHaveBeenCalledWith('snap-origin', newState);
+    });
+
+    it('throws an error on update if the new state is not plain object', async () => {
+      const clearSnapState = jest.fn().mockResolvedValueOnce(true);
+      const getSnapState = jest.fn().mockResolvedValueOnce(true);
+      const updateSnapState = jest.fn().mockResolvedValueOnce(true);
+
+      const manageStateImplementation = getManageStateImplementation({
+        clearSnapState,
+        getSnapState,
+        updateSnapState,
+      });
+      const newState = (a: unknown) => {
+        return a;
+      };
+
+      await expect(
+        manageStateImplementation({
+          context: { origin: 'snap-origin' },
+          method: '',
+          params: {
+            operation: ManageStateOperation.updateState,
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-expect-error
```

Comment:
Explain why, i.e. "invalid type for testing purposes"

Reply:
Added these explanations where needed.